annotator/src/main/java/org/cbioportal/annotator/internal/GenomeNexusImpl.java

```diff
@@ -654,16 +654,27 @@ else if (transcripts.size() > 1) {
     }
 
     private TranscriptConsequence transcriptWithMostSevereConsequence(List<TranscriptConsequence> transcripts, String mostSevereConsequence) {
+        Integer highestPriority = Integer.MAX_VALUE;
+        TranscriptConsequence highestPriorityTranscript = null;
         for (TranscriptConsequence transcript : transcripts) {
             List<String> consequenceTerms = transcript.getConsequenceTerms();
             for (String consequenceTerm : consequenceTerms) {
+                if (effectPriority.getOrDefault(consequenceTerm.toLowerCase(), Integer.MAX_VALUE) < highestPriority) {
+                    highestPriorityTranscript = transcript;
+                    highestPriority = effectPriority.getOrDefault(consequenceTerm.toLowerCase(), Integer.MAX_VALUE);
+                }
                 if (consequenceTerm.trim().equals(mostSevereConsequence)) {
                     return transcript;
```

Comment:
If it matches the most_severe_consequence, it returns that. If none of the transcripts match, it returns the transcript with the highest priority consequence instead of just returning the first one.